tests/test_cli.py

```diff
@@ -513,4 +518,54 @@ def test_license_with_file(self, fetch_database_url):
         test_filename = os.path.join(dirname, "reqs_4.txt")
         result = self.runner.invoke(cli.cli, ['license', '--key', 'foo', '--file', test_filename])
         print(result.stdout)
-        self.assertEqual(result.exit_code, 0)
\ No newline at end of file
+        self.assertEqual(result.exit_code, 0)
+
+    @patch('safety.auth.cli.get_auth_info', return_value={'email': 'test@test.com'})
+    @patch.object(Auth, 'is_valid', return_value=True)
+    @patch('safety.auth.utils.SafetyAuthSession.get_authentication_type', return_value=AuthenticationType.TOKEN)
+    @patch('builtins.input', lambda *args: '')
+    @patch('safety.safety.fetch_database', return_value={'vulnerable_packages': []})
+    def test_debug_flag(self, mock_get_auth_info, mock_is_valid, mock_get_auth_type, mock_fetch_database):
+        result = self.runner.invoke(cli.cli, ['--debug', 'scan'])
+        assert result.exit_code == 0, f"CLI exited with code {result.exit_code} and output: {result.output} and error: {result.stderr}"
+        assert "for known security issues using default" in result.output
+
+    @patch('safety.auth.cli.get_auth_info', return_value={'email': 'test@test.com'})
+    @patch.object(Auth, 'is_valid', return_value=True)
+    @patch('safety.auth.utils.SafetyAuthSession.get_authentication_type', return_value=AuthenticationType.TOKEN)
+    @patch('builtins.input', lambda *args: '')
+    @patch('safety.safety.fetch_database', return_value={'vulnerable_packages': []})
+    def test_debug_flag_with_value_1(self, mock_get_auth_info, mock_is_valid, mock_get_auth_type, mock_fetch_database):
+        sys.argv = ['safety', '--debug', '1', 'scan']
+
+        @cli.preprocess_args
+        def dummy_function():
+            pass
+
+        dummy_function()  # Call the decorated function
```

Comment:
i think lines 545 and 565 should be deleted, because it's not actually necessary to call the function in order to trigger the behavior. in fact, it's important that the behavior gets triggered *before* the function is called.

Reply:
Sounds good, fixed now!